### PR TITLE
Fix last(?) batch of malloc-NULL places

### PR DESCRIPTION
**Pull request by richsalz**
Add a script to find them in the future

### CODE REVIEW

File: crypto/lhash/lhash.c [hunk 1]
```diff
@@ -45,8 +46,10 @@ OPENSSL_LHASH *OPENSSL_LH_new(OPENSSL_LH_HASHFUNC h, OPENSSL_LH_COMPFUNC c)
 {
     OPENSSL_LHASH *ret;
 
-    if ((ret = OPENSSL_zalloc(sizeof(*ret))) == NULL)
+    if ((ret = OPENSSL_zalloc(sizeof(*ret))) == NULL) {
+        CRYPTOerr(CRYPTO_F_OPENSSL_LH_NEW, ERR_R_MALLOC_FAILURE);
```
Comment (bernd-edlinger) [inline crypto/lhash/lhash.c]: be careful here!
this lhash code is poisoned.

Comment (richsalz) [reply crypto/lhash/lhash.c]: you mean beware of err/lhash recursion?  If so, I'll comment that line out and explain why.  Or do you mean something else?

Comment (bernd-edlinger) [reply crypto/lhash/lhash.c]: Yes.  I am not 100% sure, but think I recall there is a code path form the error strings
init code which uses a hash object.

Comment (richsalz) [reply crypto/lhash/lhash.c]: updated the commit to comment out the call and explain why.

**Comment by richsalz** [comment]
Thanks for the feedback; merged.

**Pull request closed without merging**